### PR TITLE
[misc] optimize for loops

### PR DESCRIPTION
**Pull request by PulkitAgg**
This pull request will improve the performance as I am pre calculating the array length before the for loop execution
For example
```
let i, arr = [1,2,3,4,5];
for(i = 0; i< arr.length; i++) {
    console.log(i)
}
```
arr.length will execute in each iteration which slows our execution. For this I just pre calculate the length and use that variable in each iteration like
```
let i, arr = [1,2,3,4,5], arrLen = arr.length;
for(i = 0; i< arrLen; i++) {
    console.log(i)
}
```

I am also attachig the test suite case run time which is decrease by 610ms after changes

Before
Test_Case_Run_Time_Before_Code_Update

After
Test_Case_Run_Time_After_Code_Update

### CODE REVIEW

**Comment by PulkitAgg** [comment]
@marwahaha I have already signed CLA (Contributor License Agreement) but don't know why its showing in pending state. Does I have to do something else. Please help me because this is my first contribution to open source community.

**Comment by marwahaha** [comment]
Thanks for this. How did you find these cases? Did you look manually? Is there a lint rule we can set up to avoid this in the future?

**Comment by PulkitAgg** [comment]
@marwahaha I look manually for these cases. I don't have any idea about such rule but we can look for this rule.

File: src/lib/create/from-string.js [hunk 1]
```diff
@@ -63,12 +63,13 @@ export function configFromISO(config) {
         allowTime,
         dateFormat,
         timeFormat,
-        tzFormat;
+        tzFormat,
+        isoDateLen = isoDates.length,
+        isoTimeLen = isoTimes.length;
```
Comment (marwahaha) [inline src/lib/create/from-string.js:68]: Could you change these to `isoDatesLen` and `isoTimesLen`?

Comment (PulkitAgg) [reply src/lib/create/from-string.js:68]: These are just a variables for isoDates and isoTimes length which are used in 72 and 84th line

Comment (PulkitAgg) [reply src/lib/create/from-string.js]: @marwahaha I have updated these variables

**Comment by lifeart** [comment]
may be related - https://mrale.ph/blog/2014/12/24/array-length-caching.html

**Pull request merged**